### DISCUSSION TITLE
Is there something like VFileOutput for file export?

### DISCUSSION THREAD

**Discussion by bigfooted** [Kitware/trame#294] [Q&A] [answered]
Hi all,
I'd like to be able to save a file in the same way as you would open a file using VFileInput. So I  would like to navigate to a specific location and then either click on an existing file or give a new filename as input. Is something like that available?

**Comment by jourdain** [comment, marked as answer]
Actually using standard browser, you won't be able to get a file path that way (for security reason). 
If you are using Electron or Tauri, then you should have a way to grab the full path of the file. 
Otherwise, you will have to implement the [file navigation via web UI](https://github.com/Kitware/trame-pvui/blob/master/trame_pvui/widgets/filebrowser.py) using some server logic for file navigation like in the picture below.

 
<img width="1116" alt="Screen Shot 2023-07-20 at 5 54 26 PM" src="https://github.com/Kitware/trame/assets/294518/4e6deaa1-0fa8-4279-a991-6d69f333cd0e">

**Reply by bigfooted** [reply]
OK, clear. 
Instead, could I download the file, i.e. it appears in the browsers 'Downloads' list and it is saved in the Downloads folder?

**Reply by jourdain** [reply]
Yes we have 2 examples showing you how to do it. 

One using a [state variable that contains text](https://github.com/Kitware/trame/blob/master/examples/00_howdoi/download.py). And another one that [download some binary content](https://github.com/Kitware/trame/blob/master/examples/validation/core/24_vtk_download_image.py) computed on the server.

**Reply by bigfooted** [reply]
OK, thanks! This is actually closer to what I need, although the first answer is the answer to the original question :-)
